build-frontend.sh

```diff
@@ -1,6 +1,43 @@
 #!/bin/bash
 
-docker build --target build_frontend -t circuitbreaker-frontend-builder .
-CID=$(docker create circuitbreaker-frontend-builder)
-docker cp $CID:/webui-build .
-docker rm -f $CID
\ No newline at end of file
+# Function to compare version numbers
+version_greater_or_equal() {
+  printf '%s\n' "$2" "$1" | sort -V -C
+}
+
+# Function to install or update Yarn
+install_or_update_yarn() {
+  if [ "$EUID" -eq 0 ]; then
+    echo "Installing Yarn as superuser..."
+    npm install -g yarn
+  else
+    echo "Installing Yarn as current user..."
+    sudo npm install -g yarn
+  fi
+}
+
+# Check if Docker is installed
+if command -v docker >/dev/null 2>&1; then
+  docker build --target build_frontend -t circuitbreaker-frontend-builder .
+  CID=$(docker create circuitbreaker-frontend-builder)
+  docker cp $CID:/webui-build .
+  docker rm -f $CID
+else
+  # Check if Yarn is installed and if the version is greater than or equal to 1.10.0
+  if command -v yarn >/dev/null 2>&1; then
+    YARN_VERSION=$(yarn --version)
+    if version_greater_or_equal "$YARN_VERSION" "1.10.0"; then
+      echo "Yarn is installed with version $YARN_VERSION"
+    else
```

Comment:
Thanks for providing this script. I can see this working, but the downside is that this script needs to be maintained and also that local installation of tools may conflict with other builds that require different versions.

I tend more towards https://github.com/lightningequipment/circuitbreaker/issues/67#issuecomment-1495646745 where we go back to checking in the frontend, but paired with a CI check.

Reply:
Sure! Until you have a solution for #67 I suggest to leave that PR temporary open.

Reply:
Because I saw noderunners in Telegram groups who want to use the latest version of circuitbreaker on a non-docker basis but experience problems of a missing web frontend, I created this script to compile the  web frontend.

This script addresses all noderunners who are on raspibolt/ raspiblitz without docker.

Leave some comments for Joost or me if you experience problems with the script. Maybe the modified bash script gets integrated in the repo 
or  a non-docker version of circuitbreaker will be created. 


The script applies to circuitbreaker v.0.4.3
To install the modified build-frontend.sh bash script from git use following commands:

cd circuitbreaker
git fetch origin pull/69/head:patch-1
git checkout patch-1
./build-frontend.sh

Reply:
Given the seemingly persisting interest for this, I've created https://github.com/lightningequipment/circuitbreaker/pull/73. Let me know if that does the trick.